driver-core/src/main/java/com/datastax/driver/core/policies/FallthroughRetryPolicy.java

```diff
@@ -86,4 +89,67 @@ public RetryDecision onWriteTimeout(Statement statement, ConsistencyLevel cl, Wr
     public RetryDecision onUnavailable(Statement statement, ConsistencyLevel cl, int requiredReplica, int aliveReplica, int nbRetry) {
         return RetryDecision.rethrow();
     }
+
+    /**
+     * Defines whether to retry and at which consistency level on a
+     * client timeout.
+     * <p>
+     * This implementation triggers a retry on the next host in the query plan,
+     * regardless of the consistency level or the number of retries.
+     *
+     * @param statement the original query for which the consistency level cannot
+     * be achieved.
+     * @param cl the original consistency level for the operation.
+     * @param nbRetry the number of retry already performed for this operation.
+     * @return the retry decision. If {@code RetryDecision.RETHROW} is returned,
+     * an {@link OperationTimedOutException} will
+     * be thrown for the operation.
+     */
+    @Override
+    public RetryDecision onClientTimeout(Statement statement, ConsistencyLevel cl, int nbRetry) {
+        return RetryDecision.tryNextHost(cl);
```

Comment:
I wonder if this policy should return RETHROW here. It would be more consistent with the class purpose, but would break the current behavior.